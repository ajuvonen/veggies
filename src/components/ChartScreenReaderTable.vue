<script lang="ts" setup>
defineProps<{
  id: string;
  title: string;
  columnHeaders: string[];
  data: number[][];
  rowHeaders?: string[];
}>();
</script>
<template>
  <table class="sr-only" :id="id">
    <caption>
      {{
        title
      }}
    </caption>
    <thead>
      <tr>
        <td v-if="rowHeaders"></td>
        <th v-for="header in columnHeaders" :key="header">{{ header }}</th>
      </tr>
    </thead>
    <tbody>
      <template v-if="rowHeaders">
        <tr v-for="(header, headerIndex) in rowHeaders" :key="headerIndex">
          <th>{{ header }}</th>
          <td v-for="(value, dataIndex) in data[headerIndex]" :key="dataIndex">{{ value }}</td>
        </tr>
      </template>
      <tr v-else>
        <td v-for="(value, index) in data[0]" :key="index">{{ value }}</td>
      </tr>
    </tbody>
  </table>
</template>
