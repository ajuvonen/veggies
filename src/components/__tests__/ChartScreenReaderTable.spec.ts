import {describe, it, expect} from 'vitest';
import {mount} from '@vue/test-utils';
import ChartScreenReaderTable from '@/components/ChartScreenReaderTable.vue';

describe('ChartScreenReaderTable', () => {
  it('mounts', () => {
    const wrapper = mount(ChartScreenReaderTable, {
      props: {
        title: 'Amount of veggies',
        id: 'my-chart',
        columnHeaders: ['Grains', 'Vegetables', 'Roots & Bulbs'],
        data: [[1, 2, 3]],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('table').classes()).toContain('sr-only');
  });

  it('mounts with row headers', () => {
    const wrapper = mount(ChartScreenReaderTable, {
      props: {
        title: 'Amount of veggies',
        id: 'my-chart',
        columnHeaders: ['Fruits', 'Grains', 'Leafy Greens'],
        rowHeaders: ['Week 1', 'Week 2', 'Week 3'],
        data: [
          [1, 2, 3],
          [1, 1, 2],
          [3, 3, 0],
        ],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
