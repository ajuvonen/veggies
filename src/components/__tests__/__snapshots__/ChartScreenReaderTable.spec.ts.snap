// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChartScreenReaderTable > mounts 1`] = `
"<table class="sr-only" id="my-chart">
  <caption>Amount of veggies</caption>
  <thead>
    <tr>
      <!--v-if-->
      <th>Grains</th>
      <th>Vegetables</th>
      <th>Roots &amp; Bulbs</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>2</td>
      <td>3</td>
    </tr>
  </tbody>
</table>"
`;

exports[`ChartScreenReaderTable > mounts with row headers 1`] = `
"<table class="sr-only" id="my-chart">
  <caption>Amount of veggies</caption>
  <thead>
    <tr>
      <td></td>
      <th>Fruits</th>
      <th>Grains</th>
      <th>Leafy Greens</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>Week 1</th>
      <td>1</td>
      <td>2</td>
      <td>3</td>
    </tr>
    <tr>
      <th>Week 2</th>
      <td>1</td>
      <td>1</td>
      <td>2</td>
    </tr>
    <tr>
      <th>Week 3</th>
      <td>3</td>
      <td>3</td>
      <td>0</td>
    </tr>
  </tbody>
</table>"
`;
